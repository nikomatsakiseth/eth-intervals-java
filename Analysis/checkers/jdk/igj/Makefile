
.SILENT:

# Variables
CHECKERS ?= ../..
CHECKERS_JAR ?= $(CHECKERS)/checkers.jar

LANGTOOLS ?= ../../../jsr308-langtools
JAVADOCLIB ?= $(LANGTOOLS)/build/classes
JAVAC ?= $(LANGTOOLS)/dist/bin/javac -Xlint
JAVAP ?= $(LANGTOOLS)/dist/bin/javap

RTJAR ?= $(JAVA_HOME)/jre/lib/rt.jar

clean:
	rm -rf build annotated jdk.jar

build:
	mkdir -p build


############## Validate annotated jdk against missing methods ##############
SOURCES = $(shell find src/java -name '*.java')
CLASSES = $(patsubst src/%.java,build/%.class, $(SOURCES))
DIFFS = $(patsubst src/%.java,build/%.diff, $(SOURCES))

.PHONY: classes
classes: $(SOURCES)
	echo "building the jdk..."
	mkdir -p build
	$(JAVAC) -cp $(CHECKERS_JAR):$(JAVADOCLIB) -d build $?

build/%.class: src/%.java
	$(JAVAC) -sourcepath src -cp $(CHECKERS_JAR):$(JAVADOCLIB) -d build $<

build/%.expected: build/%.class
	$(JAVAP) -cp $(JAVADOCLIB) $(subst /,.,$*)  | grep public | sed 's/ native / /g' | sed 's/? extends java\/lang\/Object/?/g' | sort > $@

build/%.actual: build/%.class
	$(JAVAP) -Xbootclasspath/p:build $(subst /,.,$*)  | grep public | sed 's/ native / /g' | sed 's/? extends java\/lang\/Object/?/g' | sort > $@

build/%.diff: build/%.expected build/%.actual
	echo "validating class:" $(subst /,.,$*)
	diff $? > $@

validate: classes $(DIFFS)

############## Annotate the annotated jdk file ##############
JAIFS = $(patsubst src/%.java,build/%.jaif, $(SOURCES))
ANNOTATED = $(patsubst src/%.java,annotated/%.class, $(SOURCES))

build/%.jaif: build/%.class
	extract-annotations $<

stubs: $(JAIFS)

annotated/%.class: build/%.jaif
	cd annotated; unzip $(RTJAR) $*.class > /dev/null
	insert-annotations $@ $<

annotated: $(ANNOTATED)

jdk.jar: annotated
	cd annotated; jar cf ../jdk.jar `find . -name '*.class'`

jdk: jdk.jar
