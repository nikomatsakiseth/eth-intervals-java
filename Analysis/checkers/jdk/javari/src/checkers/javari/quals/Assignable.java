package checkers.javari.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import checkers.javari.JavariChecker;

import static java.lang.annotation.ElementType.*;

/**
 * Indicates that a field is assignable, even if it is inside a {@link ReadOnly}
 * instance.
 *
 * <p>
 *
 * This annotation is part of the Javari language.
 *
 * @see JavariChecker
 * @checker.framework.manual #javari-checker Javari Checker
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({FIELD})
public @interface Assignable {

}
