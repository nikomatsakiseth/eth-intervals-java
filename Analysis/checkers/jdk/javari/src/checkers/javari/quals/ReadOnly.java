package checkers.javari.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import checkers.javari.JavariChecker;
import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;
import static java.lang.annotation.ElementType.*;

/**
 * Indicates that, for the variable on which this annotation appears,
 * the object to which this variable refers will not be modified via
 * this reference, except its fields explicitly marked as
 * {@link Mutable}.
 *
 * <p>
 *
 * This annotation is part of the Javari language.
 *
 * @see Mutable
 * @see JavariChecker
 * @checker.framework.manual #javari-checker Javari Checker
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
//@Target({FIELD, LOCAL_VARIABLE, METHOD, PARAMETER, TYPE})
@TypeQualifier
@SubtypeOf({})
public @interface ReadOnly {

}
