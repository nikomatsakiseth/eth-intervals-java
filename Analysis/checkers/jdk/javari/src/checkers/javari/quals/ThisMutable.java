package checkers.javari.quals;

import java.lang.annotation.Target;

import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;

/**
 * An annotation used to represent a place holder immutability type, that is
 * equivalent to the ThisMutable type on the Javari typesystem.
 *
 * However, it is an implementation detail; hence, the package-scope.
 */
@TypeQualifier
@Target({}) // A programmer cannot write @ThisMutable in a program
@SubtypeOf(ReadOnly.class)
public @interface ThisMutable {

}
