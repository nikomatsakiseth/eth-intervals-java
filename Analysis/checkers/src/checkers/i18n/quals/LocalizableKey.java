package checkers.i18n.quals;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import checkers.quals.SubtypeOf;
import checkers.quals.TypeQualifier;
import checkers.quals.Unqualified;

/**
 * Indicates that the {@code String} type has been localized and
 * formatted for the target output locale
 */
@TypeQualifier
@SubtypeOf(Unqualified.class)
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface LocalizableKey { }
