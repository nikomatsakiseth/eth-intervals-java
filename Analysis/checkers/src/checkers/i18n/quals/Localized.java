package checkers.i18n.quals;

import static com.sun.source.tree.Tree.Kind.*;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import checkers.quals.*;

/**
 * Indicates that the {@code String} type has been localized and
 * formatted for the target output locale
 */
@TypeQualifier
@SubtypeOf(Unqualified.class)
@ImplicitFor( trees = {
        /* All integer literals */
        INT_LITERAL,
        LONG_LITERAL,
        FLOAT_LITERAL,
        DOUBLE_LITERAL,
        BOOLEAN_LITERAL,

        //CHAR_LITERAL,
        //STRING_LITERAL,
        //NULL_LITERAL
        }
)
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface Localized { }
