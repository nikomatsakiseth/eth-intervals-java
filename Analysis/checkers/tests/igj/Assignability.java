import checkers.igj.quals.*;

@I
public class Assignability {
    int field = 3;
    @Assignable int assignable;

    public Assignability() {
        field = 1;
        field *= 1;
        field /= 1;
        field += 1;
        field -= 1;
        field++;
        ++field;
        field--;
        --field;

        this.field = 1;
        this.field *= 1;
        this.field /= 1;
        this.field += 1;
        this.field -= 1;
        this.field++;
        ++this.field;
        this.field--;
        --this.field;

        assignable = 1;
        assignable *= 1;
        assignable /= 1;
        assignable += 1;
        assignable -= 1;
        assignable++;
        ++assignable;
        assignable--;
        --assignable;

        this.assignable = 1;
        this.assignable *= 1;
        this.assignable /= 1;
        this.assignable += 1;
        this.assignable -= 1;
        this.assignable++;
        ++this.assignable;
        this.assignable--;
        --this.assignable;

        @Mutable Assignability mutableRef = null;
        mutableRef.field = 1;
        mutableRef.field *= 1;
        mutableRef.field /= 1;
        mutableRef.field += 1;
        mutableRef.field -= 1;
        mutableRef.field++;
        ++mutableRef.field;
        mutableRef.field--;
        --mutableRef.field;
        mutableRef.assignable = 1;
        mutableRef.assignable *= 1;
        mutableRef.assignable /= 1;
        mutableRef.assignable += 1;
        mutableRef.assignable -= 1;
        mutableRef.assignable++;
        ++mutableRef.assignable;
        mutableRef.assignable--;
        --mutableRef.assignable;

        @ReadOnly Assignability readOnlyRef = (@ReadOnly Assignability)null;
        readOnlyRef.field = 1; // error
        readOnlyRef.field *= 1;    // error
        readOnlyRef.field /= 1;    // error
        readOnlyRef.field += 1;    // error
        readOnlyRef.field -= 1;    // error
        readOnlyRef.field++;   // error
        ++readOnlyRef.field;   // error
        readOnlyRef.field--;   // error
        --readOnlyRef.field;   // error
        readOnlyRef.assignable = 1;
        readOnlyRef.assignable *= 1;
        readOnlyRef.assignable /= 1;
        readOnlyRef.assignable += 1;
        readOnlyRef.assignable -= 1;
        readOnlyRef.assignable++;
        ++readOnlyRef.assignable;
        readOnlyRef.assignable--;
        --readOnlyRef.assignable;

        @Immutable Assignability immutableRef = (@Immutable Assignability)null;
        immutableRef.field = 1; // error
        immutableRef.field *= 1;    // error
        immutableRef.field /= 1;    // error
        immutableRef.field += 1;    // error
        immutableRef.field -= 1;    // error
        immutableRef.field++;   // error
        ++immutableRef.field;   // error
        immutableRef.field--;   // error
        --immutableRef.field;   // error
        immutableRef.assignable = 1;
        immutableRef.assignable *= 1;
        immutableRef.assignable /= 1;
        immutableRef.assignable += 1;
        immutableRef.assignable -= 1;
        immutableRef.assignable++;
        ++immutableRef.assignable;
        immutableRef.assignable--;
        --immutableRef.assignable;
    }

    public void readOnly() @ReadOnly {
        field = 1;  // error
        field *= 1; // error
        field /= 1; // error
        field += 1; // error
        field -= 1; // error
        field++;    // error
        ++field;    // error
        field--;    // error
        --field;    // error

        this.field = 1; // error
        this.field *= 1;    // error
        this.field /= 1;    // error
        this.field += 1;    // error
        this.field -= 1;    // error
        this.field++;   // error
        ++this.field;   // error
        this.field--;   // error
        --this.field;   // error

        assignable = 1;
        assignable *= 1;
        assignable /= 1;
        assignable += 1;
        assignable -= 1;
        assignable++;
        ++assignable;
        assignable--;
        --assignable;

        this.assignable = 1;
        this.assignable *= 1;
        this.assignable /= 1;
        this.assignable += 1;
        this.assignable -= 1;
        this.assignable++;
        ++this.assignable;
        this.assignable--;
        --this.assignable;

        @Mutable Assignability mutableRef = (@Mutable Assignability)null;
        mutableRef.field = 1;
        mutableRef.field *= 1;
        mutableRef.field /= 1;
        mutableRef.field += 1;
        mutableRef.field -= 1;
        mutableRef.field++;
        ++mutableRef.field;
        mutableRef.field--;
        --mutableRef.field;
        mutableRef.assignable = 1;
        mutableRef.assignable *= 1;
        mutableRef.assignable /= 1;
        mutableRef.assignable += 1;
        mutableRef.assignable -= 1;
        mutableRef.assignable++;
        ++mutableRef.assignable;
        mutableRef.assignable--;
        --mutableRef.assignable;

        @ReadOnly Assignability readOnlyRef = (@ReadOnly Assignability)null;
        readOnlyRef.field = 1; // error
        readOnlyRef.field *= 1;    // error
        readOnlyRef.field /= 1;    // error
        readOnlyRef.field += 1;    // error
        readOnlyRef.field -= 1;    // error
        readOnlyRef.field++;   // error
        ++readOnlyRef.field;   // error
        readOnlyRef.field--;   // error
        --readOnlyRef.field;   // error
        readOnlyRef.assignable = 1;
        readOnlyRef.assignable *= 1;
        readOnlyRef.assignable /= 1;
        readOnlyRef.assignable += 1;
        readOnlyRef.assignable -= 1;
        readOnlyRef.assignable++;
        ++readOnlyRef.assignable;
        readOnlyRef.assignable--;
        --readOnlyRef.assignable;

        @Immutable Assignability immutableRef = (@Immutable Assignability)null;
        immutableRef.field = 1; // error
        immutableRef.field *= 1;    // error
        immutableRef.field /= 1;    // error
        immutableRef.field += 1;    // error
        immutableRef.field -= 1;    // error
        immutableRef.field++;   // error
        ++immutableRef.field;   // error
        immutableRef.field--;   // error
        --immutableRef.field;   // error
        immutableRef.assignable = 1;
        immutableRef.assignable *= 1;
        immutableRef.assignable /= 1;
        immutableRef.assignable += 1;
        immutableRef.assignable -= 1;
        immutableRef.assignable++;
        ++immutableRef.assignable;
        immutableRef.assignable--;
        --immutableRef.assignable;

    }

    public void mutable() @Mutable {
        field = 1;
        field *= 1;
        field /= 1;
        field += 1;
        field -= 1;
        field++;
        ++field;
        field--;
        --field;

        this.field = 1;
        this.field *= 1;
        this.field /= 1;
        this.field += 1;
        this.field -= 1;
        this.field++;
        ++this.field;
        this.field--;
        --this.field;

        assignable = 1;
        assignable *= 1;
        assignable /= 1;
        assignable += 1;
        assignable -= 1;
        assignable++;
        ++assignable;
        assignable--;
        --assignable;

        this.assignable = 1;
        this.assignable *= 1;
        this.assignable /= 1;
        this.assignable += 1;
        this.assignable -= 1;
        this.assignable++;
        ++this.assignable;
        this.assignable--;
        --this.assignable;

        @Mutable Assignability mutableRef = (@Mutable Assignability)null;
        mutableRef.field = 1;
        mutableRef.field *= 1;
        mutableRef.field /= 1;
        mutableRef.field += 1;
        mutableRef.field -= 1;
        mutableRef.field++;
        ++mutableRef.field;
        mutableRef.field--;
        --mutableRef.field;
        mutableRef.assignable = 1;
        mutableRef.assignable *= 1;
        mutableRef.assignable /= 1;
        mutableRef.assignable += 1;
        mutableRef.assignable -= 1;
        mutableRef.assignable++;
        ++mutableRef.assignable;
        mutableRef.assignable--;
        --mutableRef.assignable;

        @ReadOnly Assignability readOnlyRef = (@ReadOnly Assignability )null;
        readOnlyRef.field = 1; // error
        readOnlyRef.field *= 1;    // error
        readOnlyRef.field /= 1;    // error
        readOnlyRef.field += 1;    // error
        readOnlyRef.field -= 1;    // error
        readOnlyRef.field++;   // error
        ++readOnlyRef.field;   // error
        readOnlyRef.field--;   // error
        --readOnlyRef.field;   // error
        readOnlyRef.assignable = 1;
        readOnlyRef.assignable *= 1;
        readOnlyRef.assignable /= 1;
        readOnlyRef.assignable += 1;
        readOnlyRef.assignable -= 1;
        readOnlyRef.assignable++;
        ++readOnlyRef.assignable;
        readOnlyRef.assignable--;
        --readOnlyRef.assignable;

        @Immutable Assignability immutableRef = (@Immutable Assignability)null;
        immutableRef.field = 1; // error
        immutableRef.field *= 1;    // error
        immutableRef.field /= 1;    // error
        immutableRef.field += 1;    // error
        immutableRef.field -= 1;    // error
        immutableRef.field++;   // error
        ++immutableRef.field;   // error
        immutableRef.field--;   // error
        --immutableRef.field;   // error
        immutableRef.assignable = 1;
        immutableRef.assignable *= 1;
        immutableRef.assignable /= 1;
        immutableRef.assignable += 1;
        immutableRef.assignable -= 1;
        immutableRef.assignable++;
        ++immutableRef.assignable;
        immutableRef.assignable--;
        --immutableRef.assignable;


    }

    public void immutable() @Immutable {
        field = 1;  // error
        field *= 1; // error
        field /= 1; // error
        field += 1; // error
        field -= 1; // error
        field++;    // error
        ++field;    // error
        field--;    // error
        --field;    // error

        this.field = 1; // error
        this.field *= 1;    // error
        this.field /= 1;    // error
        this.field += 1;    // error
        this.field -= 1;    // error
        this.field++;   // error
        ++this.field;   // error
        this.field--;   // error
        --this.field;   // error

        assignable = 1;
        assignable *= 1;
        assignable /= 1;
        assignable += 1;
        assignable -= 1;
        assignable++;
        ++assignable;
        assignable--;
        --assignable;

        this.assignable = 1;
        this.assignable *= 1;
        this.assignable /= 1;
        this.assignable += 1;
        this.assignable -= 1;
        this.assignable++;
        ++this.assignable;
        this.assignable--;
        --this.assignable;

        @Mutable Assignability mutableRef = (@Mutable Assignability)null;
        mutableRef.field = 1;
        mutableRef.field *= 1;
        mutableRef.field /= 1;
        mutableRef.field += 1;
        mutableRef.field -= 1;
        mutableRef.field++;
        ++mutableRef.field;
        mutableRef.field--;
        --mutableRef.field;
        mutableRef.assignable = 1;
        mutableRef.assignable *= 1;
        mutableRef.assignable /= 1;
        mutableRef.assignable += 1;
        mutableRef.assignable -= 1;
        mutableRef.assignable++;
        ++mutableRef.assignable;
        mutableRef.assignable--;
        --mutableRef.assignable;

        @ReadOnly Assignability readOnlyRef = (@ReadOnly Assignability)null;
        readOnlyRef.field = 1; // error
        readOnlyRef.field *= 1;    // error
        readOnlyRef.field /= 1;    // error
        readOnlyRef.field += 1;    // error
        readOnlyRef.field -= 1;    // error
        readOnlyRef.field++;   // error
        ++readOnlyRef.field;   // error
        readOnlyRef.field--;   // error
        --readOnlyRef.field;   // error
        readOnlyRef.assignable = 1;
        readOnlyRef.assignable *= 1;
        readOnlyRef.assignable /= 1;
        readOnlyRef.assignable += 1;
        readOnlyRef.assignable -= 1;
        readOnlyRef.assignable++;
        ++readOnlyRef.assignable;
        readOnlyRef.assignable--;
        --readOnlyRef.assignable;

        @Immutable Assignability immutableRef = (@Immutable Assignability)null;
        immutableRef.field = 1; // error
        immutableRef.field *= 1;    // error
        immutableRef.field /= 1;    // error
        immutableRef.field += 1;    // error
        immutableRef.field -= 1;    // error
        immutableRef.field++;   // error
        ++immutableRef.field;   // error
        immutableRef.field--;   // error
        --immutableRef.field;   // error
        immutableRef.assignable = 1;
        immutableRef.assignable *= 1;
        immutableRef.assignable /= 1;
        immutableRef.assignable += 1;
        immutableRef.assignable -= 1;
        immutableRef.assignable++;
        ++immutableRef.assignable;
        immutableRef.assignable--;
        --immutableRef.assignable;

    }

    public void mutableByDefault() {
        field = 1;
        field *= 1;
        field /= 1;
        field += 1;
        field -= 1;
        field++;
        ++field;
        field--;
        --field;

        this.field = 1;
        this.field *= 1;
        this.field /= 1;
        this.field += 1;
        this.field -= 1;
        this.field++;
        ++this.field;
        this.field--;
        --this.field;

        assignable = 1;
        assignable *= 1;
        assignable /= 1;
        assignable += 1;
        assignable -= 1;
        assignable++;
        ++assignable;
        assignable--;
        --assignable;

        this.assignable = 1;
        this.assignable *= 1;
        this.assignable /= 1;
        this.assignable += 1;
        this.assignable -= 1;
        this.assignable++;
        ++this.assignable;
        this.assignable--;
        --this.assignable;

        @Mutable Assignability mutableRef = (@Mutable Assignability)null;
        mutableRef.field = 1;
        mutableRef.field *= 1;
        mutableRef.field /= 1;
        mutableRef.field += 1;
        mutableRef.field -= 1;
        mutableRef.field++;
        ++mutableRef.field;
        mutableRef.field--;
        --mutableRef.field;
        mutableRef.assignable = 1;
        mutableRef.assignable *= 1;
        mutableRef.assignable /= 1;
        mutableRef.assignable += 1;
        mutableRef.assignable -= 1;
        mutableRef.assignable++;
        ++mutableRef.assignable;
        mutableRef.assignable--;
        --mutableRef.assignable;

        @ReadOnly Assignability readOnlyRef = (@ReadOnly Assignability)null;
        readOnlyRef.field = 1; // error
        readOnlyRef.field *= 1;    // error
        readOnlyRef.field /= 1;    // error
        readOnlyRef.field += 1;    // error
        readOnlyRef.field -= 1;    // error
        readOnlyRef.field++;   // error
        ++readOnlyRef.field;   // error
        readOnlyRef.field--;   // error
        --readOnlyRef.field;   // error
        readOnlyRef.assignable = 1;
        readOnlyRef.assignable *= 1;
        readOnlyRef.assignable /= 1;
        readOnlyRef.assignable += 1;
        readOnlyRef.assignable -= 1;
        readOnlyRef.assignable++;
        ++readOnlyRef.assignable;
        readOnlyRef.assignable--;
        --readOnlyRef.assignable;

        @Immutable Assignability immutableRef = (@Immutable Assignability)null;
        immutableRef.field = 1; // error
        immutableRef.field *= 1;    // error
        immutableRef.field /= 1;    // error
        immutableRef.field += 1;    // error
        immutableRef.field -= 1;    // error
        immutableRef.field++;   // error
        ++immutableRef.field;   // error
        immutableRef.field--;   // error
        --immutableRef.field;   // error
        immutableRef.assignable = 1;
        immutableRef.assignable *= 1;
        immutableRef.assignable /= 1;
        immutableRef.assignable += 1;
        immutableRef.assignable -= 1;
        immutableRef.assignable++;
        ++immutableRef.assignable;
        immutableRef.assignable--;
        --immutableRef.assignable;

    }


}
