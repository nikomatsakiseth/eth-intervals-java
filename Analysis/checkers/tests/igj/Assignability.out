Assignability.java:70: (assignability.invalid)
Assignability.java:71: (assignability.invalid)
Assignability.java:72: (assignability.invalid)
Assignability.java:73: (assignability.invalid)
Assignability.java:74: (assignability.invalid)
Assignability.java:75: (assignability.invalid)
Assignability.java:76: (assignability.invalid)
Assignability.java:77: (assignability.invalid)
Assignability.java:78: (assignability.invalid)
Assignability.java:90: (assignability.invalid)
Assignability.java:91: (assignability.invalid)
Assignability.java:92: (assignability.invalid)
Assignability.java:93: (assignability.invalid)
Assignability.java:94: (assignability.invalid)
Assignability.java:95: (assignability.invalid)
Assignability.java:96: (assignability.invalid)
Assignability.java:97: (assignability.invalid)
Assignability.java:98: (assignability.invalid)
Assignability.java:111: (assignability.invalid)
Assignability.java:112: (assignability.invalid)
Assignability.java:113: (assignability.invalid)
Assignability.java:114: (assignability.invalid)
Assignability.java:115: (assignability.invalid)
Assignability.java:116: (assignability.invalid)
Assignability.java:117: (assignability.invalid)
Assignability.java:118: (assignability.invalid)
Assignability.java:119: (assignability.invalid)
Assignability.java:121: (assignability.invalid)
Assignability.java:122: (assignability.invalid)
Assignability.java:123: (assignability.invalid)
Assignability.java:124: (assignability.invalid)
Assignability.java:125: (assignability.invalid)
Assignability.java:126: (assignability.invalid)
Assignability.java:127: (assignability.invalid)
Assignability.java:128: (assignability.invalid)
Assignability.java:129: (assignability.invalid)
Assignability.java:172: (assignability.invalid)
Assignability.java:173: (assignability.invalid)
Assignability.java:174: (assignability.invalid)
Assignability.java:175: (assignability.invalid)
Assignability.java:176: (assignability.invalid)
Assignability.java:177: (assignability.invalid)
Assignability.java:178: (assignability.invalid)
Assignability.java:179: (assignability.invalid)
Assignability.java:180: (assignability.invalid)
Assignability.java:192: (assignability.invalid)
Assignability.java:193: (assignability.invalid)
Assignability.java:194: (assignability.invalid)
Assignability.java:195: (assignability.invalid)
Assignability.java:196: (assignability.invalid)
Assignability.java:197: (assignability.invalid)
Assignability.java:198: (assignability.invalid)
Assignability.java:199: (assignability.invalid)
Assignability.java:200: (assignability.invalid)
Assignability.java:275: (assignability.invalid)
Assignability.java:276: (assignability.invalid)
Assignability.java:277: (assignability.invalid)
Assignability.java:278: (assignability.invalid)
Assignability.java:279: (assignability.invalid)
Assignability.java:280: (assignability.invalid)
Assignability.java:281: (assignability.invalid)
Assignability.java:282: (assignability.invalid)
Assignability.java:283: (assignability.invalid)
Assignability.java:295: (assignability.invalid)
Assignability.java:296: (assignability.invalid)
Assignability.java:297: (assignability.invalid)
Assignability.java:298: (assignability.invalid)
Assignability.java:299: (assignability.invalid)
Assignability.java:300: (assignability.invalid)
Assignability.java:301: (assignability.invalid)
Assignability.java:302: (assignability.invalid)
Assignability.java:303: (assignability.invalid)
Assignability.java:318: (assignability.invalid)
Assignability.java:319: (assignability.invalid)
Assignability.java:320: (assignability.invalid)
Assignability.java:321: (assignability.invalid)
Assignability.java:322: (assignability.invalid)
Assignability.java:323: (assignability.invalid)
Assignability.java:324: (assignability.invalid)
Assignability.java:325: (assignability.invalid)
Assignability.java:326: (assignability.invalid)
Assignability.java:328: (assignability.invalid)
Assignability.java:329: (assignability.invalid)
Assignability.java:330: (assignability.invalid)
Assignability.java:331: (assignability.invalid)
Assignability.java:332: (assignability.invalid)
Assignability.java:333: (assignability.invalid)
Assignability.java:334: (assignability.invalid)
Assignability.java:335: (assignability.invalid)
Assignability.java:336: (assignability.invalid)
Assignability.java:379: (assignability.invalid)
Assignability.java:380: (assignability.invalid)
Assignability.java:381: (assignability.invalid)
Assignability.java:382: (assignability.invalid)
Assignability.java:383: (assignability.invalid)
Assignability.java:384: (assignability.invalid)
Assignability.java:385: (assignability.invalid)
Assignability.java:386: (assignability.invalid)
Assignability.java:387: (assignability.invalid)
Assignability.java:399: (assignability.invalid)
Assignability.java:400: (assignability.invalid)
Assignability.java:401: (assignability.invalid)
Assignability.java:402: (assignability.invalid)
Assignability.java:403: (assignability.invalid)
Assignability.java:404: (assignability.invalid)
Assignability.java:405: (assignability.invalid)
Assignability.java:406: (assignability.invalid)
Assignability.java:407: (assignability.invalid)
Assignability.java:482: (assignability.invalid)
Assignability.java:483: (assignability.invalid)
Assignability.java:484: (assignability.invalid)
Assignability.java:485: (assignability.invalid)
Assignability.java:486: (assignability.invalid)
Assignability.java:487: (assignability.invalid)
Assignability.java:488: (assignability.invalid)
Assignability.java:489: (assignability.invalid)
Assignability.java:490: (assignability.invalid)
Assignability.java:502: (assignability.invalid)
Assignability.java:503: (assignability.invalid)
Assignability.java:504: (assignability.invalid)
Assignability.java:505: (assignability.invalid)
Assignability.java:506: (assignability.invalid)
Assignability.java:507: (assignability.invalid)
Assignability.java:508: (assignability.invalid)
Assignability.java:509: (assignability.invalid)
Assignability.java:510: (assignability.invalid)
